src/dsymbol/tests.d

```diff
@@ -484,6 +484,49 @@ unittest
 	assert(a.data.ptr == b.data.ptr);
 }
 
+// this is for testing if breadcrumbs are properly generating the calltip
+// we only process symbols with suffixes (ptr, array, aa)
+unittest
+{
+	ModuleCache cache = ModuleCache(theAllocator);
+
+	writeln("Running calltip tests...");
+	auto source = q{ struct A {int a;}
+		A value; A* ptr = &a; A** ptrptr = &ptr; A[] arr; A*[] arrptr; A[int] map; A*[int] mapptr;
+
+		// builtin type test
+		int* intptr;
+		int[] intarr;
+		int[int] intaa;
+	};
+	auto pair = generateAutocompleteTrees(source, cache);
+	auto A = pair.symbol.getFirstPartNamed(internString("value"));
+	auto B = pair.symbol.getFirstPartNamed(internString("ptr"));
+	auto C = pair.symbol.getFirstPartNamed(internString("ptrptr"));
+	auto D = pair.symbol.getFirstPartNamed(internString("arr"));
+	auto E = pair.symbol.getFirstPartNamed(internString("arrptr"));
+	auto F = pair.symbol.getFirstPartNamed(internString("map"));
+	auto G = pair.symbol.getFirstPartNamed(internString("mapptr"));
+
+	auto H = pair.symbol.getFirstPartNamed(internString("intptr"));
+	auto I = pair.symbol.getFirstPartNamed(internString("intarr"));
+	auto J = pair.symbol.getFirstPartNamed(internString("intaa"));
+
+	// empty because it's not a pointer/array/aa
+	assert(A.type.callTip == "");
```

Comment:
for consistency I think it would make more sense if all variables had callTips and not just pointer/array/aa here.

Reply:
Sent the commit, but that would require some changes in DCD/serve-d

https://github.com/Pure-D/serve-d/blob/b26abad4cce04f54f105825d9e4a03ff0ecab644/source/served/commands/complete.d#L954

Need to remove this, and put the logic in DCD  here: 

https://github.com/dlang-community/DCD/blob/02acaa534b9be65142aed7b202a6a8d5524abf2a/src/dcd/server/autocomplete/util.d#L756-L768


I'll send a PRs

Reply:
Welp, DCD needs the changes from DSymbol first

So this needs to get merged first